Sources/Bond/Collections/ObservableArray.swift

```diff
@@ -295,30 +296,30 @@ extension MutableObservableArray {
 }
 
 extension MutableObservableArray where Item: Equatable {
-  
+
   public func replace(with array: [Item], performDiff: Bool) {
-    if performDiff {
-      lock.lock()
+    lock.lock(); defer { lock.unlock() }
 
+    if performDiff {
       let diff = self.array.extendedDiff(array)
+      let patch = diff.patch(from: self.array, to: array)
```

Comment:
Sadly, my hunch is correct. 

UIKIt allows you (within the confines of `beginUpdate()`/`endUpdate()`) to address elements by their pre-modification indices: 

```
move(from: 1, to: 5)
move(from: 2, to: 6)
```

AppKit expects you to use the post-modification sequence. The same sequence as above would need to be represented as:

```
move(from: 1, to: 5)
move(from: 1, to: 5)
```

I'll have a think about how to handle this - obviously I don't want to break UIKit to address AppKit's needs.